### PR TITLE
Fix date escape sequences

### PR DESCRIPTION
**Pull request by toptalo**
Add tests and fix issue with escape sequences in date filter

For Issue #563

### CODE REVIEW

**Comment by toptalo** [comment]
Seems like Issue #557 here

File: src/twig.filters.js [hunk 1]
```diff
@@ -284,10 +284,11 @@ module.exports = function (Twig) {
 
             return obj;
         },
+
         date: function(value, params) {
             var date = Twig.functions.date(value);
             var format = params && params.length ? params[0] : 'F j, Y H:i';
-            return Twig.lib.date(format, date);
+            return Twig.lib.date(format.replace(/\\\\/g,'\\'), date);
```
Comment (willrowe) [inline src/twig.filters.js]: Add a space between the arguments in the `format.replace` call.

File: test/templates/escape-date-format.twig [hunk 2]
```diff
@@ -0,0 +1 @@
+{{ "Tue Aug 14 08:52:15 +0000 2007"|date("F jS \\a\\t g:ia") }}
```
Comment (willrowe) [inline test/templates/escape-date-format.twig]: Simplify the input date string to `1970-01-01 00:00:00`.

File: test/test.filters.js [hunk 3]
```diff
@@ -266,19 +266,38 @@ describe("Twig.js Filters ->", function() {
...
+
         it("should recognize string date formats", function() {
             var template = twig({data: '{{ "Tue Aug 14 08:52:15 +0000 2007"|date("d/m/Y @ H:i:s") }}'})
                 , date = new Date(1187081535000); // 14/08/2007 @ 04:52:15 EST
 
             template.render().should.equal( stringDate(date) );
         });
 
+        it("should escape words and characters in the date format (twig:data)]", function () {
+            var template = twig({data: '{{ "Tue Aug 14 08:52:15 +0000 2007"|date("F jS \\a\\t g:ia") }}'});
```
Comment (willrowe) [inline test/test.filters.js]: Simplify the input date string to `1970-01-01 00:00:00`.

File: test/test.filters.js [hunk 4]
```diff
@@ -266,19 +266,38 @@ describe("Twig.js Filters ->", function() {
...
+
+            template.render().should.equal( "August 14th at 12:52pm" );
```
Comment (willrowe) [inline test/test.filters.js]: Change this to equal `January 1st at 12:00am`.

File: test/test.filters.js [hunk 5]
```diff
@@ -266,19 +266,38 @@ describe("Twig.js Filters ->", function() {
...
+
+        it("should escape words and characters in the date format (twig:ref)]", function () {
+            twig({
+                id: 'escape-date-format',
+                path: 'test/templates/escape-date-format.twig',
+                async: false
+            });
+
+            // Load the template
+            twig({ref: 'escape-date-format'}).render({}).should.equal("August 14th at 12:52pm");
```
Comment (willrowe) [inline test/test.filters.js]: Change this to equal `January 1st at 12:00am`.

**Pull request merged**